it/src/main/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManager.java

```diff
@@ -0,0 +1,239 @@
+
+  private static final int DEFAULT_ACK_DEADLINE_SECONDS = 600;
+  private static final String RESOURCE_NAME_SEPARATOR = "-";
+
+  private final String testId;
+  private final String projectId;
+  private final CredentialsProvider credentialsProvider;
+  private final TopicAdminClient topicAdminClient;
+  private final SubscriptionAdminClient subscriptionAdminClient;
+
+  private final List<TopicName> createdTopics;
+  private final List<SubscriptionName> createdSubscriptions;
+
+  private DefaultPubsubResourceManager(Builder builder) throws IOException {
+    this(
+        builder.testName,
+        builder.projectId,
+        builder.credentialsProvider,
+        TopicAdminClient.create(
+            TopicAdminSettings.newBuilder()
+                .setCredentialsProvider(builder.credentialsProvider)
+                .build()),
+        SubscriptionAdminClient.create(
+            SubscriptionAdminSettings.newBuilder()
+                .setCredentialsProvider(builder.credentialsProvider)
+                .build()));
+  }
+
+  @VisibleForTesting
+  DefaultPubsubResourceManager(
+      String testName,
+      String projectId,
+      CredentialsProvider credentialsProvider,
+      TopicAdminClient topicAdminClient,
+      SubscriptionAdminClient subscriptionAdminClient) {
+    this.projectId = projectId;
+    this.testId = PubsubUtils.createTestId(testName);
+    this.credentialsProvider = credentialsProvider;
+    this.topicAdminClient = topicAdminClient;
+    this.subscriptionAdminClient = subscriptionAdminClient;
+    this.createdTopics = Collections.synchronizedList(new ArrayList<>());
+    this.createdSubscriptions = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  public static Builder builder(String testName, String projectId) {
+    checkArgument(!Strings.isNullOrEmpty(testName), "testName can not be null or empty");
+    checkArgument(!projectId.isEmpty(), "projectId can not be empty");
+    return new Builder(testName, projectId);
+  }
+
+  @Override
+  public TopicName createTopic(String topicName) {
+    checkArgument(!topicName.isEmpty(), "topicName can not be empty");
+    checkIsUsable();
+
+    LOG.info("Creating topic '{}'...", topicName);
+
+    Topic topic = topicAdminClient.createTopic(getTopicName(topicName));
+    TopicName reference = PubsubUtils.toTopicName(topic);
+    createdTopics.add(reference);
+
+    LOG.info("Topic '{}' was created successfully!", topicName);
+
+    return reference;
+  }
+
+  @Override
+  public SubscriptionName createSubscription(String topicName, String subscriptionName) {
+    checkArgument(!topicName.isEmpty(), "topicName can not be empty");
+    checkArgument(!subscriptionName.isEmpty(), "subscriptionName can not be empty");
+    checkIsUsable();
+
+    return createSubscription(getTopicName(topicName), subscriptionName);
+  }
+
+  @Override
+  public SubscriptionName createSubscription(TopicName topicName, String subscriptionName) {
```

Comment:
I'm not sure if this is supposed to be a helper or alternative public method. (Maybe just a helper for now?)

Either way, the checks are a bit inconsistent. We double-check the subscription name if coming through the other method but don't call `checkIsUsable` here, which seems more critical here than in the other method.